Sources/ActionHolder.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+public struct ActionHolder <A: Equatable> {
+  let origin: A
+  var modified: A
+  
+  public static func create(action: A) -> ActionHolder<A> {
+    return ActionHolder(origin: action,
+                        modified: action)
+  }
+  
+  func isOrigin(action: Any) -> Bool {
+    guard let castedAction = action as? A else {
```

Comment:
In my mind It doesn't matter since we anyway cast `action` parameter to ActionHolder's generic type `A` that has Equatable requirement. If we decide to use some specific type requirement for `action` parameter, Tit might be useful to create a type alias for action type that currently is defined as Equatable to avoid semantical confusion. I mean something kinda
 `typealias JanetAction = Equatable`